src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageHandlerMethodSelector.cs

```diff
@@ -113,5 +159,38 @@ private static int GetScore(HandlerMethodDescriptor descriptor)
                 return 0;
             }
         }
+
+        private static string GetHandlerName(PageContext context)
+        {
+            var handlerName = Convert.ToString(context.RouteData.Values[Handler]);
+            if (!string.IsNullOrEmpty(handlerName))
+            {
+                return handlerName;
+            }
+
+            if (context.HttpContext.Request.Query.TryGetValue(Handler, out StringValues queryValues))
+            {
+                return queryValues[0];
+            }
+
+            return null;
+        }
+
+        private static string GetFuzzyMatchHttpMethod(PageContext context)
+        {
+            var httpMethod = context.HttpContext.Request.Method;
+
+            // All of the 'safe' HTTP methods map to GET
+            if (string.Equals("GET", httpMethod, StringComparison.OrdinalIgnoreCase) ||
```

Comment:
Should we make this mapping a first class feature? We can configure the defaults to match what we want, but it does open up the possibility of doing interesting things like do a fetch with `Delete` method.

Reply:
What would making it configurable help with? What do you imagine it would do?